src/wristpy/processing/analytics.py

```diff
@@ -308,41 +308,80 @@ def _compute_abs_diff_mean_anglez(
             measurements=absolute_diff, time=anglez_epoch1.time[1:]
         )
 
-    def _find_periods(
-        self, window_measurement: models.Measurement
-    ) -> List[Tuple[datetime, datetime]]:
-        """Find periods where window_measurement is equal to 1.
 
-        This is a helper function for the _find_onset_wakeup_times to
-        find periods where either the spt_window or sib_periods are equal to 1.
+def _find_periods(
+    window_measurement: models.Measurement,
+) -> List[Tuple[datetime, datetime]]:
+    """Find periods where window_measurement is equal to 1.
 
-        Args:
-            window_measurement: the Measurement instance, intended to be
-                either the spt_window or sib_period.
+    This is a helper function for the _find_onset_wakeup_times to
+    find periods where either the spt_window or sib_periods are equal to 1.
 
-        Returns:
-            A list of tuples, where each tuple contains the start and end times of
-            a period. For isolated ones the function returns the same start
-            and end time. The list is sorted by time.
-        """
-        edge_detection = np.convolve([1, 3, 1], window_measurement.measurements, "same")
-        single_one = np.nonzero(edge_detection == 3)[0]
-
-        single_periods = [
-            (window_measurement.time.item(idx), window_measurement.time.item(idx))
-            for idx in single_one
-        ]
-
-        blocked_one_edge = np.nonzero(edge_detection == 4)[0]
-        block_pairs = np.reshape(blocked_one_edge, (-1, 2))
-        block_periods = [
-            (window_measurement.time.item(idx[0]), window_measurement.time.item(idx[1]))
-            for idx in block_pairs
-        ]
-        all_periods = single_periods + block_periods
-        all_periods.sort()
-
-        return all_periods
+    Args:
+        window_measurement: the Measurement instance, intended to be
+            either the spt_window or sib_period.
+
+    Returns:
+        A list of tuples, where each tuple contains the start and end times of
+        a period. For isolated ones the function returns the same start
+        and end time. The list is sorted by time.
+    """
+    edge_detection = np.convolve([1, 3, 1], window_measurement.measurements, "same")
+    single_one = np.nonzero(edge_detection == 3)[0]
+
+    single_periods = [
+        (window_measurement.time.item(idx), window_measurement.time.item(idx))
+        for idx in single_one
+    ]
+
+    blocked_one_edge = np.nonzero(edge_detection == 4)[0]
+    block_pairs = np.reshape(blocked_one_edge, (-1, 2))
+    block_periods = [
+        (window_measurement.time.item(idx[0]), window_measurement.time.item(idx[1]))
+        for idx in block_pairs
+    ]
+    all_periods = single_periods + block_periods
+    all_periods.sort()
+
+    return all_periods
+
+
+def remove_nonwear_from_sleep(
+    non_wear_array: models.Measurement,
+    sleep_windows: List,
```

Comment:
Huh I forgot it there but it was in the description in the docstring. Fixed

Reply:
Oh, now I remember. It's because SleepWindow can be `datetime | List` so the current version leads to a mypy error later. Added a check to ensure that it is datetime first (aka not empty sleepwindow)